client/src/components/forms/MailingList/slides/Feedback.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { Dispatch, SetStateAction } from 'react';
+import /*type*/ { WithStyles, Theme } from '@material-ui/core/styles';
+import { withStyles, createStyles } from '@material-ui/core/styles';
+import { Typography, FormGroup, FormControlLabel, Checkbox, TextField } from '@material-ui/core';
+import { EVENT } from '../types';
+
+const styles = (theme: Theme) => createStyles({
+  container: {
+    display: 'flex',
+    flexDirection: 'column',
+  },
+  textContainer: {
+    display: 'flex',
+    flexDirection: 'column',
+    marginBottom: theme.spacing(2)
+  },
+  text: {
+    color: theme.palette.text.primary
+  },
+  eventsContainer: {
+    marginBottom: theme.spacing(3)
+  },
+  feedback: {
+    width: '90%'
+  }
+});
+
+interface Props extends WithStyles<typeof styles> {
+  events: EVENT[];
+  setEvents: Dispatch<SetStateAction<EVENT[]>>;
+  feedback: string;
+  setFeedback: Dispatch<SetStateAction<string>>;
+}
+
+function Feedback({ classes, events, setEvents, feedback, setFeedback }: Props) {
+  const toggleEvent = (event: EVENT) => {
+    const index = events.indexOf(event);
+    if(index > -1) {
+      events.splice(index, 1);
+    } else {
+      events.push(event);
+    }
+    setEvents(events);
+  }
+
+  return (
+    <div className={classes.container}>
+      <div className={classes.textContainer}>
+        <Typography variant="h4" className={classes.text}>
+            What events are you interested in?
+        </Typography>
+        <Typography variant="body1" className={classes.text}>
+          {"Are there any ACS events that you're looking forward to this term?"}
+        </Typography>
+      </div>
+      <div className={classes.eventsContainer}>
+        {Object.values(EVENT).map(event => (
+          <FormGroup key={event}>
+            <FormControlLabel
+              control={
+                <Checkbox
+                  size='small'
+                  checked={events.indexOf(event) > -1}
```

Comment:
This is the boy right here. Have to find a different way to check this.

Reply:
Weird bug... I would think this would work. And because you aren't unmounting and remounting anything it's super strange that it works when you navigate away and back 👀. Not sure if you have any leads already but maybe it's something to do with the translation not displaying the checkbox checked correctly.

Reply:
Nevermind what I sent above. It's because you are pushing or splicing and setting the same array object instead of creating a new one to set. React thinks the array hasn't changed since it has the same reference so it doesn't rerender. Clicking the buttons updates another state which causes the rerender.

Reply:
Daaamn thats actually sick. It's because the reference didn't change. Thanks Josh!

Reply:
I think I might change the state to an object instead because now we have a `title` and a `description` for each event.

(You don't have to care about the description from an api perspective. As long as each event has a unique identifier (title) we should be good with that for now)